Code/Tools/AssetProcessor/native/utilities/Builder.h

```diff
@@ -0,0 +1,165 @@
+/*
+ * Copyright (c) Contributors to the Open 3D Engine Project.
+ * For complete copyright and license terms please see the LICENSE at the root of this distribution.
+ *
+ * SPDX-License-Identifier: Apache-2.0 OR MIT
+ *
+ */
+#pragma once
+
+#include <AzCore/std/string/string.h>
+#include <AzCore/std/parallel/binary_semaphore.h>
+#include <utilities/assetUtils.h>
+#include <AzFramework/Process/ProcessWatcher.h>
+#include <AzFramework/Process/ProcessCommunicatorTracePrinter.h>
+
+namespace AssetProcessor
+{
+    //! Enum used to indicate the purpose of a builder which may result in special handling
+    enum class BuilderPurpose
+    {
+        CreateJobs,
+        ProcessJob,
+        Registration
+    };
+
+    enum class BuilderRunJobOutcome
+    {
+        Ok,
+        LostConnection,
+        ProcessTerminated,
+        JobCancelled,
+        ResponseFailure,
+        FailedToDecodeResponse,
+        FailedToWriteDebugRequest
+    };
+
+    //! Wrapper for managing a single builder process and sending job requests to it
+    class Builder
+    {
+        friend class BuilderManager;
+        friend struct BuilderRef;
+        friend class BuilderList;
+
+    public:
+        Builder(const AssetUtilities::QuitListener& quitListener, AZ::Uuid uuid)
+            : m_uuid(uuid)
+            , m_quitListener(quitListener)
+        {
+        }
+        virtual ~Builder() = default;
+
+        // Disable copy and move (can't move a semaphore)
+        AZ_DISABLE_COPY_MOVE(Builder);
+
+        //! Returns true if the builder has a valid connection id and, if there is a process associated, the process is running
+        bool IsValid() const;
+
+        //! Returns true if the builder has a process watcher and the process is running OR does not have a process watcher.  Returns false
+        //! otherwise
+        bool IsRunning(AZ::u32* exitCode = nullptr) const;
```

Comment:
As called out previously, consider UNIX style version of this call and split into IsRunning() GetErrorCode()